simulator/src/test/java/org/hiero/block/simulator/TestUtils.java

```diff
@@ -48,4 +49,8 @@ public static int findFreePort() throws IOException {
             return socket.getLocalPort();
         }
     }
+
+    public static String getAbsoluteFolder(String relativePath) {
```

Comment:
Q: Should this have a null guard with a test also?
```suggestion
    public static String getAbsoluteFolder(@NonNull String relativePath) {
```

Reply:
You are right. If relativePath is null, calling Paths.get(relativePath) will throw a NullPointerException. Adding @NonNull is suitable for this case.